server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -327,6 +327,60 @@ protected void doClose() {
         }
     }
 
+    @Override
+    public void executeConsistentStateUpdate(Function<RepositoryData, ClusterStateUpdateTask> createUpdateTask,
+                                             Consumer<Exception> onFailure) {
+        threadPool.generic().execute(new AbstractRunnable() {
+            @Override
+            protected void doRun() {
+                final RepositoryMetadata repositoryMetadataStart = metadata;
+                getRepositoryData(ActionListener.wrap(repositoryData ->
+                        clusterService.submitStateUpdateTask("consistent state update", new ClusterStateUpdateTask() {
```

Comment:
`"consistent state update"`. Can you pass in a more descriptive state update message?

Reply:
Also, can we add the timeout here now?

Reply:
Sure added the timeout, passing a source now + took the opportunity to also pass the task priority along now.